### PR TITLE
Add endpoints for creating and listing auth factors

### PR DESCRIPTION
**Pull request by alisherry**
## Description
Add endpoints for 

1) `enroll_auth_factor` to enroll a user in an auth factor
Introduces a joint object `AuthenticationFactorAndChallenge` for the response 

2) `list_auth_factors` to list a user's auth factors. 

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ X ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: lib/workos/user_management.rb [hunk 1]
```diff
@@ -463,6 +463,60 @@ def authenticate_user_with_code(code:, client_id:, ip_address: nil, user_agent:
         )
         WorkOS::AuthenticationResponse.new(response.body)
       end
+
+      # Enroll a user into an authentication factor.
+      #
+      # @param [String] id The id for the user.
```
Comment (willmanduffy) [inline lib/workos/user_management.rb:469]: We should make this be `user_id`

File: lib/workos/user_management.rb [hunk 2]
```diff
@@ -463,6 +463,60 @@ def authenticate_user_with_code(code:, client_id:, ip_address: nil, user_agent:
...
+            auth: true,
+          ),
+        )
+
+        WorkOS::AuthenticationFactorAndChallenge.new(response.body)
+      end
+
+      # Get all auth factors for a user
+      #
+      # @param [String] id The id for the user.
```
Comment (willmanduffy) [inline lib/workos/user_management.rb:493]: This should also be `user_id`

File: lib/workos/user_management.rb [hunk 3]
```diff
@@ -463,6 +463,60 @@ def authenticate_user_with_code(code:, client_id:, ip_address: nil, user_agent:
...
+        params(
+          id: String,
+        ).returns(WorkOS::AuthenticationFactorAndChallenge)
+      end
+      def enroll_auth_factor(id:)
+        response = execute_request(
+          request: post_request(
+            path: "/users/#{id}/auth/factors",
+            body: {
+              type: 'topd',
```
Comment (willmanduffy) [inline lib/workos/user_management.rb:482]:
```suggestion
              type: 'totp',
```

File: lib/workos/user_management.rb [hunk 4]
```diff
@@ -463,6 +463,60 @@ def authenticate_user_with_code(code:, client_id:, ip_address: nil, user_agent:
...
+
+        parsed_response = JSON.parse(response.body)
+
+        auth_factors = parsed_response.map do |auth_factor|
+          ::WorkOS::Factor.new(auth_factor.to_json)
+        end
+
+        WorkOS::Types::ListStruct.new(
+          data: auth_factors,
+          list_metadata: {},
```
Comment (willmanduffy) [inline lib/workos/user_management.rb:517]: Is there no pagination on this endpoint?

Comment (alisherry) [reply lib/workos/user_management.rb:517]: It doesn't seem like we include a list_metadata object in the endpoint based on what I was checking but I'll double check

**Review by adamzinder — APPROVED** [review]

**Pull request merged**